src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -80,4 +112,23 @@ public function itsCurrentVariantShouldBeNamed(string $variantName): void
             )
         );
     }
+
+    private function hasProductWithPrice(array $resource, int $price, ?string $productCode = null): bool
+    {
+        foreach ($resource as $product) {
+            if ($productCode && $product['code'] !== $productCode) {
```

Comment:
Is there any case that $productCode may not exist? Just asking @SirDomin ?

Reply:
if you are checking collection, you should specify product code to check wanted product from the list. Otherwise we get only one product, so we dont need to search for it.

Reply:
`$productCode` should have explicit compare to `null` (to skip casting). Also `$productCode` check seems to always return false (if product code is equal to null, we will always return false), therefore we can extract this check before the foreach loop. Or even better remove nullable from this value